src/main/kotlin/Main.kt

```diff
@@ -36,22 +43,31 @@ fun main(args: Array<String>) {
 
 fun getEnv(e : String, default: String? = null) : String = System.getenv()[e] ?: default ?: throw RuntimeException("Missing environment variable $e and no default value is given.")
 
-private val producer = ProducerBuilder.ofByteArray
-    .option("client.id", "leia")
-    .create()
+private fun mkProducer() =
+    ProducerBuilder.ofByteArray
+        .option("client.id", "leia")
+        .create()
 
 private val logger = KotlinLogging.logger("process-request")
 
 private val genericHeaders: Map<String, String> = mapOf(
     "Server" to "Zensum/Leia"
 )
 
-fun server(port: Int) = embeddedServer(Netty, port) {
-    val routes: Map<String, TopicRouting> = getRoutes()
+fun leia(producer: Producer<String, ByteArray>,
+         routes: Map<String, TopicRouting>,
+         jwtProvider: JWTProvider? = null,
+         installPrometheus: Boolean = true): Application.() -> Unit = {
     install(SentryFeature)
-    install(PrometheusFeature)
+    if (installPrometheus) {
```

Comment:
I can understand why this is needed from a testing perspective (I think), but it kind of feel like we have a boolean flag like `isTest` now. Is it possible to replace `installPrometheus: Boolean = true` with a list of features to install instead or a lambda block of features, so we can do the same thing for Sentry and Health if needed (I guess it isn't needed tough since they are always enabled, but in the future we may want to install more features that requires a similar behaviour when testing). So we would have an argument `features: Application.() -> Unit` and then install with `features()` in the start of the leia function. I understand that some of the resources needed in the specific scope can be an issue, like `jwtProvider` being needed for JWTFeature.

Just want to offer an alternative solution, this probably works good as well ;)

Reply:
I think we should consider this out of the scope for this PR. Lets consider this function once we override two middleware. Even then I don't think this level of configuration makes sense from a testing PoV as we want it to be as realistic as possible. Maybe we should fix this in the prometheus feature but I don't know how.

Reply:
> Maybe we should fix this in the prometheus feature but I don't know how.

It is possible to uninstall a feature, I don't know if we can use that somehow. Like
```
if(featureIsInstalled)
    uninstall(PrometheusFeature)
```

Reply:
Yes but the prometheus container registers stuff, and registering them twice results in an error. Didn't know uninstall existed, though :)